pkg/oc/bootstrap/docker/up.go

```diff
@@ -876,7 +876,7 @@ func (c *CommonStartConfig) determineServerIP(out io.Writer) (string, []string,
 
 // updateNoProxy will add some default values to the NO_PROXY setting if they are not present
 func (c *CommonStartConfig) updateNoProxy() {
-	values := []string{"127.0.0.1", c.ServerIP, "localhost", openshift.RegistryServiceIP, openshift.ServiceCatalogServiceIP, "172.30.0.0/8"}
+	values := []string{"127.0.0.1", c.ServerIP, "localhost", openshift.ServiceCatalogServiceIP, openshift.RegistryServiceClusterIP}
```

Comment:
service catalog service is hard coded?  This just won't work right long term.

@pmorie @jpeeler @csrwng why is this hardcoded?  It's impractical for every service to be whitelisted here.

Reply:
it would be great if docker support CIDR for no_proxy... or some sort of wildcard

Reply:
@deads2k I can't answer for the service catalog service ip, but for the registry service ip, we needed to include it in the no_proxy list of ip's so we could push to it. And yes, unfortunately, docker won't honor a CIDR in the no_proxy list.

Reply:
@bparees did the cluster up work for service catalog. Is this still needed for catalog, Ben?

Reply:
presumably this was done to ensure brokers could call the service catalog to register with it.
or perhaps to ensure the web console could call the service catalog.  I don't have any particular recollection about hitting a problem that resulted in me putting this there, so it may in part have simply been a monkey-see-monkey-do.

I don't remember the specifics, but it has nothing to do w/ docker.  The problem is linux in general does not support subnet ranges in a no_proxy env var.

so it may be "impractical" but this is what we do today even in non-cluster up clusters (namely, ensure that the master api endpoint is in the no_proxy list.  Whether we ensure the service catalog is or not, i'm not sure).

Moving to using domains in no_proxy is likely the only real path forward, as long as we are sure everyone trying to contact services is doing so by a service hostname, not by service ip.